src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -60,6 +60,117 @@ public void Construction_Properties()
         }
 
         [Test]
+        public void Equals_EqualValues()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end = new LocalDate(2001, 6, 19);
+            var interval1 = new DateInterval(start, end, false);
+            var interval2 = new DateInterval(start, end, false);
+            Assert.AreEqual(interval1, interval2);
+            Assert.AreEqual(interval1.GetHashCode(), interval2.GetHashCode());
+            Assert.IsTrue(interval1 == interval2);
+            Assert.IsFalse(interval1 != interval2);
+            Assert.IsTrue(interval1.Equals(interval2)); // IEquatable implementation
+        }
+
+        [Test]
+        public void Equals_DifferentStart()
+        {
+            LocalDate start1 = new LocalDate(2000, 1, 1);
+            LocalDate start2 = new LocalDate(2000, 1, 2);
+            LocalDate end = new LocalDate(2001, 6, 19);
+            var interval1 = new DateInterval(start1, end, false);
+            var interval2 = new DateInterval(start2, end, false);
+            Assert.AreNotEqual(interval1, interval2);
+            Assert.AreNotEqual(interval1.GetHashCode(), interval2.GetHashCode());
```

Comment:
There's no guarantee, but if the test fails, the chances of it being due to a bug are extremely high.